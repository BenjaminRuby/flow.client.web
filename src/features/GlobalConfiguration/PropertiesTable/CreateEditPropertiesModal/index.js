import React, { Component } from "react";
import PropTypes from "prop-types";
import ModalWrapper from "@boomerang/boomerang-components/lib/Modal";
import ModalFlow from "@boomerang/boomerang-components/lib/ModalFlow";
import { Button } from "carbon-components-react";
import CreateEditPropertiesContent from "./CreateEditPropertiesContent";
import { Add16 } from "@carbon/icons-react";

class CreateEditPropertiesModal extends Component {
  static propTypes = {
    addPropertyInStore: PropTypes.func.isRequired,
    isOpen: PropTypes.bool,
    isEdit: PropTypes.bool,
    property: PropTypes.object,
    properties: PropTypes.array,
    handleEditClose: PropTypes.func,
    updatePropertyInStore: PropTypes.func.isRequired
  };

  render() {
    const {
      addPropertyInStore,
      isEdit,
      isOpen,
      handleEditClose,
      property,
      properties,
      updatePropertyInStore
    } = this.props;

    /**
     * arrays of values for making the key unique
     * filter out own value if editing a property, pass through all if creating
     */
    let propertyKeys = [];
    if (Array.isArray(properties)) {
      propertyKeys = properties.map(propertyObj => propertyObj.key);
      if (isEdit && property) {
        propertyKeys = propertyKeys.filter(propertyItem => propertyItem !== property.key);
      }
    }

    return (
      <ModalWrapper
        isOpen={isOpen}
        modalProps={{ shouldCloseOnOverlayClick: false }}
        ModalTrigger={() =>
          !isEdit ? (
            <Button iconDescription="Create Property" renderIcon={Add16} size="field">
              Create Property
            </Button>
          ) : null
        }
        modalContent={closeModal => (
          <ModalFlow
            headerTitle={isEdit && property ? `EDIT ${property.label.toUpperCase()}` : "CREATE PROPERTY"}
            closeModal={isEdit ? handleEditClose : closeModal}
            confirmModalProps={{
              affirmativeAction: isEdit ? handleEditClose : closeModal,
              subTitleTop: "Your property will not be saved",
              theme: "bmrg-flow"
            }}
            theme="bmrg-flow"
          >
            <CreateEditPropertiesContent
              addPropertyInStore={addPropertyInStore}
              handleEditClose={handleEditClose}
              isEdit={isEdit}
              property={property}
              propertyKeys={propertyKeys}
              updatePropertyInStore={updatePropertyInStore}
            />
          </ModalFlow>
        )}
      />
    );
  }
}

export default CreateEditPropertiesModal;
