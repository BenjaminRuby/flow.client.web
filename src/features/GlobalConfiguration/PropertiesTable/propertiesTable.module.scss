@import "~@boomerang/boomerang-styles/lib/colors";
@import "~Styles/variables";

.search {
  margin-bottom: 1.31rem;
  width: 20rem;
}
.header {
  display: flex;
  justify-content: space-between;
}
.tableContainer {
  margin: 2.5rem 2.5rem 4.5rem 2.5rem;
}

.tableHeadHeader {
  color: $tab-blue;
  font-size: inherit;
  font-family: inherit;
  height: 3rem;

  button {
    width: 100% !important;
  }

  &.label {
    width: 20%;
  }

  &.key {
    width: 20%;
  }

  &.description {
    width: 25%;
  }

  &.value {
    width: 25%;
  }

  &.secured {
    width: 5%;
  }
}

.tableHeadRow {
  background-color: $cool-white-1;
}

.tableBody {
  background-color: #f2f2f2;
  color: $darkest-gray;
  font-weight: 300;
}

.tableCell {
  align-items: center;
  display: flex;
  height: 3.625rem;
  line-height: 1.2;
  overflow-y: auto;
  padding-left: 0.75rem;
  width: 100%;

  .tableSecured {
    height: 1.25rem;
    width: 1.25rem;
  }

  .secured {
    path {
      fill: $green-50;
    }
  }

  .unsecured {
    path {
      fill: $red-50;
    }
  }
}

.tableTextarea {
  background: inherit;
  color: $black;
  margin-left: 0.25rem;
  max-width: 100%;
  resize: none;
  text-align: left;
  white-space: normal;
  width: 90%;
  word-break: break-all;
}
