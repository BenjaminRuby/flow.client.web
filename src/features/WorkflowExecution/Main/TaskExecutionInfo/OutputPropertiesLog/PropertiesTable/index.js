import React, { Component } from "react";
import PropTypes from "prop-types";
import Table from "@boomerang/boomerang-components/lib/Table";
import NoDisplay from "@boomerang/boomerang-components/lib/NoDisplay";
import "./styles.scss";

class PropertiesTable extends Component {
  static propTypes = {
    data: PropTypes.array.isRequired
  };

  render() {
    const columns = [
      {
        Header: "Property",
        id: "key",
        accessor: "key"
      },
      {
        Header: "Value",
        id: "value",
        accessor: "value",
        Cell: row => <span style={{ width: "18rem", whiteSpace: "normal" }}>{row.value}</span>
      }
    ];

    if (!this.props.data || !this.props.data.length) {
      return (
        <div className="c-properties-table">
          <NoDisplay text="No properties to display" />
        </div>
      );
    }
    return (
      <div className="c-properties-table">
        <Table
          data={this.props.data}
          columns={columns}
          showPagination={false}
          minRows={this.props.data.length}
          noDataText="No data to display"
          theme="bmrg-flow"
          className="b-properties-table"
          style={{ whiteSpace: "unset" }}
          defaultSorting={[
            {
              id: "key",
              desc: false
            }
          ]}
        />
      </div>
    );
  }
}

export default PropertiesTable;
