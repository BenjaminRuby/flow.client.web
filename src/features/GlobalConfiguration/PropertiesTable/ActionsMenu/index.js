import React, { useState } from "react";
import { OverflowMenu, OverflowMenuItem } from "carbon-components-react";
import ConfirmModal from "@boomerang/boomerang-components/lib/ConfirmModal";
import AlertModalWrapper from "@boomerang/boomerang-components/lib/AlertModal";
import CreateEditPropertiesModal from "../CreateEditPropertiesModal";
import "./actionMenu.module.scss";

const OverflowMenuComponent = ({ property, properties, deleteProperty, addPropertyInStore, updatePropertyInStore }) => {
  const [deleteModalIsOpen, setDeleteModalIsOpen] = useState(false);
  const [editModalIsOpen, setEditModalIsOpen] = useState(false);

  const menuOptions = [
    {
      itemText: "Edit",
      onClick: () => setEditModalIsOpen(true)
    },
    {
      itemText: "Delete",
      onClick: () => setDeleteModalIsOpen(true),
      hasDivider: true,
      isDelete: true
    }
  ];

  const handleEditClose = () => {
    setEditModalIsOpen(false);
  };

  return (
    <>
      <OverflowMenu
        floatingMenu
        ariaLabel="Overflow menu"
        iconDescription="Overflow menu icon"
        data-testid="configuration-property-table-overflow-menu"
      >
        {menuOptions.map((option, index) => (
          <OverflowMenuItem key={index} {...option} />
        ))}
      </OverflowMenu>
      {editModalIsOpen && (
        <CreateEditPropertiesModal
          isOpen
          isEdit
          property={property}
          properties={properties}
          addPropertyInStore={addPropertyInStore}
          updatePropertyInStore={updatePropertyInStore}
          handleEditClose={handleEditClose}
        />
      )}
      {deleteModalIsOpen && (
        <AlertModalWrapper
          isOpen
          modalContent={closeModal => (
            <ConfirmModal
              closeModal={() => {
                setDeleteModalIsOpen(false);
              }}
              affirmativeAction={() => {
                deleteProperty(property);
                setDeleteModalIsOpen(false);
              }}
              title={`DELETE ${property.label.toUpperCase()}?`}
              subTitleTop="It will be gone. Forever."
              negativeText="NO"
              affirmativeText="YES"
              theme="bmrg-white"
            />
          )}
        />
      )}
    </>
  );
};

export default OverflowMenuComponent;
