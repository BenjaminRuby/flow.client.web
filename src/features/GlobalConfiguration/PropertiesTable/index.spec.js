import React from "react";
import PropertiesTable from ".";
import renderer from "react-test-renderer";

jest.mock("./ActionsMenu", () => "ActionsMenu");

const mockfn = jest.fn();
const props = {
  properties: [{ id: "1", key: "test.key", value: "testing" }],
  addPropertyInStore: mockfn,
  deletePropertyInStore: mockfn,
  updatePropertyInStore: mockfn
};

describe("PropertiesTable --- Snapshot Test", () => {
  it("Capturing Snapshot of PropertiesTable", () => {
    const component = renderer.create(<PropertiesTable {...props} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
