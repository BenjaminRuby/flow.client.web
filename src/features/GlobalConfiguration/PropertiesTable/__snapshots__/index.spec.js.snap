// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertiesTable --- Snapshot Test Capturing Snapshot of PropertiesTable 1`] = `
Array [
  <div
    className="container"
  >
    <div
      className="content"
    >
      <div
        className="info"
      >
        <h1
          className="title"
        >
          Configuration
        </h1>
        <p
          className="description"
        >
          Set global properties that are accessible in flow processes.
        </p>
      </div>
      <section
        className="buttons"
      />
    </div>
  </div>,
  <div
    className="tableContainer"
  >
    <div
      className="header"
    >
      <div
        className="bx--search bx--search--xl search"
      >
        <svg
          aria-hidden={true}
          className="bx--search-magnifier"
          focusable="false"
          height={16}
          preserveAspectRatio="xMidYMid meet"
          role="img"
          style={
            Object {
              "willChange": "transform",
            }
          }
          viewBox="0 0 16 16"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 14.3L10.7 10c1.9-2.3 1.6-5.8-.7-7.7S4.2.7 2.3 3 .7 8.8 3 10.7c2 1.7 5 1.7 7 0l4.3 4.3.7-.7zM2 6.5C2 4 4 2 6.5 2S11 4 11 6.5 9 11 6.5 11 2 9 2 6.5z"
          />
        </svg>
        <label
          className="bx--label"
          htmlFor="properties-table-search"
        >
          Search
        </label>
        <input
          className="bx--search-input"
          id="properties-table-search"
          onChange={[Function]}
          placeholder="Search"
          type="text"
        />
        <button
          aria-label="Clear search input"
          className="bx--search-close bx--search-close--hidden"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-label="Clear search input"
            focusable="false"
            height={20}
            preserveAspectRatio="xMidYMid meet"
            role="img"
            style={
              Object {
                "willChange": "transform",
              }
            }
            viewBox="0 0 32 32"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4l6.6 6.6L8 22.6 9.4 24l6.6-6.6 6.6 6.6 1.4-1.4-6.6-6.6L24 9.4z"
            />
          </svg>
        </button>
      </div>
      <div
        onClick={[Function]}
        onKeyPress={[Function]}
        style={
          Object {
            "width": "fit-content",
          }
        }
      >
        <button
          className="bx--btn bx--btn--primary"
          disabled={false}
          tabIndex={0}
          type="button"
        >
          Create Property
        </button>
      </div>
    </div>
    <div
      className="bx--data-table-container"
    >
      <table
        className="bx--data-table bx--data-table--no-border"
        zebra={false}
      >
        <thead>
          <tr
            className="tableHeadRow"
          >
            <th
              className="tableHeadHeader label"
              id="label"
              scope="col"
            >
              <span
                className="bx--table-header-label"
              >
                Label
              </span>
            </th>
            <th
              className="tableHeadHeader key"
              id="key"
              scope="col"
            >
              <span
                className="bx--table-header-label"
              >
                Key
              </span>
            </th>
            <th
              className="tableHeadHeader description"
              id="description"
              scope="col"
            >
              <span
                className="bx--table-header-label"
              >
                Description
              </span>
            </th>
            <th
              className="tableHeadHeader value"
              id="value"
              scope="col"
            >
              <span
                className="bx--table-header-label"
              >
                Value
              </span>
            </th>
            <th
              className="tableHeadHeader secured"
              id="secured"
              scope="col"
            >
              <span
                className="bx--table-header-label"
              >
                Secured
              </span>
            </th>
            <th
              className="tableHeadHeader actions "
              id="actions "
              scope="col"
            >
              <span
                className="bx--table-header-label"
              >
                Actions
              </span>
            </th>
          </tr>
        </thead>
        <tbody
          aria-live="polite"
          className="tableBody"
        >
          <tr
            data-testid="configuration-property-table-row"
          >
            <td
              style={
                Object {
                  "padding": "0",
                }
              }
            >
              <div
                className="tableCell"
              >
                <p
                  className="tableTextarea"
                >
                  ---
                </p>
              </div>
            </td>
            <td
              style={
                Object {
                  "padding": "0",
                }
              }
            >
              <div
                className="tableCell"
              >
                <p
                  className="tableTextarea"
                >
                  test.key
                </p>
              </div>
            </td>
            <td
              style={
                Object {
                  "padding": "0",
                }
              }
            >
              <div
                className="tableCell"
              >
                <p
                  className="tableTextarea"
                >
                  ---
                </p>
              </div>
            </td>
            <td
              style={
                Object {
                  "padding": "0",
                }
              }
            >
              <div
                className="tableCell"
              >
                <p
                  className="tableTextarea"
                >
                  testing
                </p>
              </div>
            </td>
            <td
              style={
                Object {
                  "padding": "0",
                }
              }
            >
              <div
                className="tableCell"
              >
                <svg
                  alt="unsecured"
                  aria-hidden={true}
                  className="tableSecured unsecured"
                  focusable="false"
                  height={32}
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "willChange": "transform",
                    }
                  }
                  viewBox="0 0 32 32"
                  width={32}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4l6.6 6.6L8 22.6 9.4 24l6.6-6.6 6.6 6.6 1.4-1.4-6.6-6.6L24 9.4z"
                  />
                </svg>
              </div>
            </td>
            <td
              style={
                Object {
                  "padding": "0",
                }
              }
            >
              <div
                className="tableCell"
              >
                <ActionsMenu
                  addPropertyInStore={[Function]}
                  deleteProperty={[Function]}
                  properties={
                    Array [
                      Object {
                        "id": "1",
                        "key": "test.key",
                        "value": "testing",
                      },
                    ]
                  }
                  property={
                    Object {
                      "id": "1",
                      "key": "test.key",
                      "value": "testing",
                    }
                  }
                  updatePropertyInStore={[Function]}
                />
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      className="bx--pagination"
      disabled={false}
    >
      <div
        className="bx--pagination__left"
      >
        <label
          className="bx--pagination__text"
          htmlFor="bx-pagination-select-1"
          id="bx-pagination-select-1-count-label"
        >
          Items per page:
        </label>
        <div
          className="bx--form-item"
        >
          <div
            className="bx--select bx--select--inline bx--select__item-count"
          >
            <div
              className="bx--select-input--inline__wrapper"
            >
              <div
                className="bx--select-input__wrapper"
                data-invalid={null}
              >
                <select
                  className="bx--select-input"
                  id="bx-pagination-select-1"
                  onChange={[Function]}
                  value={10}
                >
                  <option
                    className="bx--select-option"
                    disabled={false}
                    hidden={false}
                    value={10}
                  >
                    10
                  </option>
                  <option
                    className="bx--select-option"
                    disabled={false}
                    hidden={false}
                    value={25}
                  >
                    25
                  </option>
                  <option
                    className="bx--select-option"
                    disabled={false}
                    hidden={false}
                    value={50}
                  >
                    50
                  </option>
                </select>
                <svg
                  aria-label="open list of options"
                  className="bx--select__arrow"
                  focusable="false"
                  height={16}
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  style={
                    Object {
                      "willChange": "transform",
                    }
                  }
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 11L3 6l.7-.7L8 9.6l4.3-4.3.7.7z"
                  />
                  <title>
                    open list of options
                  </title>
                </svg>
              </div>
            </div>
          </div>
        </div>
        <span
          className="bx--pagination__text"
        >
          1–1 of 1 items
        </span>
      </div>
      <div
        className="bx--pagination__right"
      >
        <div
          className="bx--form-item"
        >
          <div
            className="bx--select bx--select--inline bx--select__page-number"
          >
            <label
              className="bx--label bx--visually-hidden"
              htmlFor="bx-pagination-select-3"
            >
              Page number, of 1 pages
            </label>
            <div
              className="bx--select-input--inline__wrapper"
            >
              <div
                className="bx--select-input__wrapper"
                data-invalid={null}
              >
                <select
                  className="bx--select-input"
                  id="bx-pagination-select-3"
                  onChange={[Function]}
                  value={1}
                >
                  <option
                    className="bx--select-option"
                    disabled={false}
                    hidden={false}
                    value={1}
                  >
                    1
                  </option>
                </select>
                <svg
                  aria-label="open list of options"
                  className="bx--select__arrow"
                  focusable="false"
                  height={16}
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  style={
                    Object {
                      "willChange": "transform",
                    }
                  }
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 11L3 6l.7-.7L8 9.6l4.3-4.3.7.7z"
                  />
                  <title>
                    open list of options
                  </title>
                </svg>
              </div>
            </div>
          </div>
        </div>
        <span
          className="bx--pagination__text"
        >
          of 1 pages
        </span>
        <button
          aria-label="Previous page"
          className="bx--pagination__button bx--pagination__button--backward bx--pagination__button--no-index"
          disabled={true}
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            focusable="false"
            height={24}
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "willChange": "transform",
              }
            }
            viewBox="0 0 32 32"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 23l-8-7 8-7v14z"
            />
          </svg>
        </button>
        <button
          aria-label="Next page"
          className="bx--pagination__button bx--pagination__button--forward bx--pagination__button--no-index"
          disabled={true}
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            focusable="false"
            height={24}
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "willChange": "transform",
              }
            }
            viewBox="0 0 32 32"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13 9l8 7-8 7V9z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
]
`;
